File: rxgooglemap/src/main/java/com/sdoward/rxgooglemap/GroundOverlayClickFunc.java
package com.sdoward.rxgooglemap;

import com.google.android.gms.maps.*;
import com.google.android.gms.maps.model.*;

import io.reactivex.Observable;
import io.reactivex.ObservableEmitter;
import io.reactivex.ObservableOnSubscribe;
import io.reactivex.functions.Function;

class GroundOverlayClickFunc implements Function<GoogleMap, Observable<GroundOverlay>> {

    @Override
    public Observable<GroundOverlay> apply(final GoogleMap googleMap) {
        return Observable.create(new ObservableOnSubscribe<GroundOverlay>() {
            @Override
            public void subscribe(final ObservableEmitter<GroundOverlay> emitter) {
                googleMap.setOnGroundOverlayClickListener(
                        new GoogleMap.OnGroundOverlayClickListener() {
                            @Override
                            public void onGroundOverlayClick(GroundOverlay groundOverlay) {
                                emitter.onNext(groundOverlay);
                            }
                        });
            }
        });
    }

}

File: rxgooglemap/src/main/java/com/sdoward/rxgooglemap/GroundOverlayClickFunc.kt
package com.sdoward.rxgooglemap

import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.model.GroundOverlay
import io.reactivex.Observable
import io.reactivex.functions.Function

class GroundOverlayClickFunc : Function<GoogleMap, Observable<GroundOverlay>> {

    override fun apply(t: GoogleMap): Observable<GroundOverlay> {
        return Observable.create { subscriber ->
            t.setOnGroundOverlayClickListener {
                subscriber.onNext(it)
            }
        }
    }

}